### PR TITLE
(feat) Allow person searching by patient identifier when defining relationships

### PR DESCRIPTION
**Pull request by icrc-psousa**
## Summary
This PR introduces the capability to search for individuals based on patient attributes, such as the patient ID, when defining relationships within the patient registration page.
This is accomplished by first querying the patient's endpoint and then merging the results with the persons endpoint, instead of solely querying the persons endpoint. This enhancement enables the search function for patient attributes that are not originally available in the persons' endpoint.

### CODE REVIEW

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx [hunk 1]
```diff
@@ -184,9 +182,23 @@ export function usePatientPhoto(patientUuid: string): UsePatientPhotoResult {
...
   const abortController = new AbortController();
 
-  return openmrsFetch(`/ws/rest/v1/person?q=${query}`, {
+  const patientsRes = await openmrsFetch(`/ws/rest/v1/patient?q=${query}`, {
     signal: abortController.signal,
   });
+
+  const personsRes = await openmrsFetch(`/ws/rest/v1/person?q=${query}`, {
+    signal: abortController.signal,
+  });
```
Comment (ibacher) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx:191]: Could we do this with a ` const [patientsRes, personsRes] = await Promise.all([fetch1, fetch2])`? That way we can run both fetches in actual parallel.

```
await fn();
await fn2();
```

Is semantically equivalent to:

```
Promise.resolve(fn()).then(Promise.resolve(fn2())).then(/* remainder of function */)
```

File: packages/esm-patient-registration-app/src/patient-registration/section/patient-relationships/relationships-section.component.tsx [hunk 2]
```diff
@@ -64,9 +64,7 @@ const RelationshipView: React.FC<RelationshipViewProps> = ({
   );
 
   const searchPerson = async (query: string) => {
-    const abortController = new AbortController();
```
Comment (ibacher) [inline packages/esm-patient-registration-app/src/patient-registration/section/patient-relationships/relationships-section.component.tsx:67]: I'd keep the abortController here and pass it as an argument to fetch person (it shouldn't be here, but that's because it should be defined further up in the component).

File: packages/esm-service-queues-app/src/patient-queue-metrics/clinic-metrics.resource.tsx [hunk 3]
```diff
@@ -21,12 +21,19 @@ export function useActiveVisits() {
...
+    const patientUUID = visit.patient?.uuid;
+    const isToday = dayjs(visit.startDatetime).isToday();
+    if (patientUUID && isToday) {
+      uniquePatientUUIDs.add(patientUUID);
+    }
+  });
 
   return {
-    activeVisitsCount: activeVisitsCount,
+    activeVisitsCount: uniquePatientUUIDs.size,
```
Comment (ibacher) [inline packages/esm-service-queues-app/src/patient-queue-metrics/clinic-metrics.resource.tsx:36]: Could we move this stuff into a separate PR? Doesn't seem related to the searching stuff.

File: packages/esm-patient-registration-app/src/routes.json [hunk 4]
```diff
@@ -45,6 +45,13 @@
       "online": true,
       "offline": true
     },
+    {
```
Comment (ibacher) [inline packages/esm-patient-registration-app/src/routes.json:48]: Ditto... I think this actually is a different PR?

Comment (icrc-psousa) [reply packages/esm-patient-registration-app/src/routes.json]: This is already in main branch. Something went wrong when merging it here. It's fixed now!
